bindings/generate_bindings.py

```diff
@@ -0,0 +1,653 @@
+        # (simplifying assumptions; if broken we need more logic)
+        responses = {**self.responses}
+        default = responses.pop("default")
+        assert isinstance(default, Ref) and default.name == "runtimeError", (
+            self.name,
+            default,
+        )
+
+        if len(responses) == 1:
+            returntype = next(iter(responses.values()))
+            returntypestr = returntype.annotation()
+        else:
+            returntypes = set(r.annotation(prequoted=True) for r in responses.values())
+            returntypestr = '"Union[' + ", ".join(sorted(returntypes)) + ']"'
+        assert len(responses) == 1, (self.name, responses)
+
+        out += [f") -> {returntypestr}:"]
+
+        # Function body.
+        path_params = sorted(p for p in self.params if self.params[p].where == "path")
+        body_params = sorted(p for p in self.params if self.params[p].where == "body")
+        query_params = sorted(p for p in self.params if self.params[p].where == "query")
+
+        pathstr = f'"{self.path}"'
+        if path_params:
+            # Happily, we can just generate an f-string based on the path swagger gives us.
+            pathstr = "f" + pathstr
+
+        if query_params:
+            out += ["    _params = {"]
+            for p in query_params:
+                param = self.params[p]
+                value = ""
+                if param.type.need_parse():
+                    value = (
+                        f"{param.type.dump(str(param.name))} if {param.name} else None"
+                    )
+                else:
+                    value = f"{param.name}"
+                out += [f'        "{self.params[p].serialized_name}": {value},']
+            out += ["    }"]
+        else:
+            out += ["    _params = None"]
+
+        if "body" in self.params:
+            bodystr = self.params["body"].dump()
+        else:
+            bodystr = "None"
+        out += ["    _resp = session._do_request("]
+        out += [f'        method="{self.method.upper()}",']
+        out += [f"        path={pathstr},"]
+        out += ["        params=_params,"]
+        out += [f"        json={bodystr},"]
+        out += ["        data=None,"]
+        out += ["        headers=None,"]
+        out += ["        timeout=None,"]
+        out += ["    )"]
+        for expect, returntype in responses.items():
+            out += [f"    if _resp.status_code == {expect}:"]
+            if returntype.isnone():
+                out += ["        return"]
+            else:
+                out += [f'        return {returntype.load("_resp.json()")}']
+        out += [f'    raise APIHttpError("{self.method}_{self.name}", _resp)']
+
+        return "\n".join(out)
+
+
+def classify_type(enums: dict, path: str, schema: dict) -> TypeAnno:
+    # enforce valid jsonschema:
+    assert isinstance(schema, dict), (path, schema)
+    if "enum" in schema:
+        name = enums[hash(json.dumps(schema["enum"]))]
```

Comment:
>  only the dictionary is able to resolve conflicts.

which conflicts are you thinking about? the possibility that the hash of two different (serialized) enums is the same? or how the keys are mapped to the underlying buckets for a dictionary

Reply:
basically allowing the hash collision resolution strategy of python's builtin dict which I don't know what it is but it has some strategy right?

Reply:
> for my understanding which conflict are you thinking about? the possibility that the hash of two different (serialized) enums is the same? or how the keys are mapped to the underlying buckets for a dictionary

Suppose that `hash()` has a conflict where `{1,2}` and `{3,4}` both hash to the same number, say 7.

When you store `mydict[hash({1,2})] = 1` and then you store `mydict[hash({3,4})]` = 2, then you'll have a map with only one value.  If you store `mydict[{1,2}) = 1` and then `mydict[{3,4}] = 2` then you'll have a map with two values.  There's a hash collision either way, but if you let the dictionary call `hash()` (as it will under the hood) it will know that even though the two entries would normally go into the same bucket they are not the same.  When you call `hash()` yourself and use that as the key, then the dictionary will just see two identical keys and they'll overwrite each other.

The chances that you have a collision with `hash()` is really low but it's also pointless to write extra code which introduces a possible bug.